src/util/test-utils.tsx

```diff
@@ -45,56 +47,82 @@ export const authUser = {
 
 export const mockAppContext: IAppContext = {
   user: authUser,
-  infoButtonData: [{ id: 'info-button-id', content: 'Info button content', title: 'Title' }],
-  appVersions: {
-    serverVersion: 'server-test-123',
-    appVersion: 'client-test-123',
-  },
-  themes: Object.keys(themesLookup),
-  selectedTheme: 'light',
-  setTheme: () => {},
 };
 
-const renderWithUserContext = (ui, { user = authUser, ...options } = {}) => {
+const renderWithUserContext = (
+  ui,
+  { user = authUser, initialStates = new Array(), ...options } = {}
+) => {
   mockAppContext.user = user;
   const Wrapper = (props) => {
     return (
-      <ThemeProvider theme={themesLookup[defaultTheme]}>
-        <AppContext.Provider value={mockAppContext} {...props} />
-      </ThemeProvider>
+      <RecoilRoot
+        initializeState={(snap) => {
+          snap.set(userState, user);
+          initialStates.forEach((s: { state: any; value: any }) => snap.set(s.state, s.value));
+        }}
+      >
+        <ThemeProvider theme={themesLookup[defaultTheme]}>
+          <AppContext.Provider value={mockAppContext} {...props} />
+        </ThemeProvider>
+      </RecoilRoot>
     );
   };
   return testingLibraryRender(ui, { wrapper: Wrapper, ...options });
 };
 
-const renderWithAppContext = (ui, { appContext = mockAppContext, ...options } = {}) => {
+const renderWithAppContext = (
+  ui,
+  { appContext = mockAppContext, initialStates = new Array(), ...options } = {}
+) => {
   const Wrapper = (props) => {
     return (
-      <ThemeProvider theme={themesLookup[defaultTheme]}>
-        <AppContext.Provider value={appContext} {...props} />
-      </ThemeProvider>
+      <RecoilRoot
+        initializeState={(snap) => {
+          snap.set(userState, authUser);
+          initialStates.forEach((s: { state: any; value: any }) => snap.set(s.state, s.value));
```

Comment:
This is a way to set/mock the initial state for any given RecoilState object (ie. plannerItemState, buttonDataState, etc).. `snap.set` is provided by RecoilRoot to initialize the states arbitrarily.. couldn't figure out a decent way to make it Typescript friendly either.